### PR TITLE
Upgrade Helix lib to Apache Helix 1.0.0

### PR DESCRIPTION
**Pull request by jsjtzyy**
1. upgraded Helix lib to latest version(1.0.0)
2. fixed test failure and conflicts due to new changes in Helix lib.

### CODE REVIEW

**Comment by jsjtzyy** [comment]
Upgrade to latest version http://helix.apache.org/1.0.0-docs/index.html
No production code changes in this PR.

File: ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixAdmin.java [hunk 1]
```diff
@@ -579,6 +600,16 @@ public void dropResource(String clusterName, String resourceName) {
     throw new IllegalStateException("Not implemented");
   }
 
+  @Override
+  public void addCloudConfig(String clusterName, CloudConfig cloudConfig) {
+
```
Comment (lightningrob) [inline ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixAdmin.java:605]: Curious why you throw IllegalStateException in the other methods but not these?

File: ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixDataAccessor.java [hunk 2]
```diff
@@ -170,19 +192,7 @@ public boolean removeProperty(PropertyKey key) {
...
-        liveInstance.setSessionId(SESSION_ID);
-        result.put(instance, (T) liveInstance);
-      }
-    } else if (key.toString().equals(INSTANCECONFIG_PATH)) {
-      for (InstanceConfig config : mockHelixAdmin.getInstanceConfigs(clusterName)) {
-        result.put(config.getInstanceName(), (T) config);
-      }
-    }
-    return result;
+    return new HashMap<>();
```
Comment (lightningrob) [inline ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixDataAccessor.java:195]: Minor: Collections.emptyMap();

**Comment by jsjtzyy** [comment]
> LGTM. Is there something specific we want from the new version?

According to its release notes, following bugs have been fixed:
```
Fix the ConcurrentModificationException in ClusterEvent.java (#785)
Generate cancellation message for currentState=null desiredState=DROPPED (#831)
Fix the concurrent modification error happens during the HelixManager initHandlers() call (#904)
Fix the scheduling decision for multiple currentStates (#923)
Fix ZkHelixPropertyStore loses Zookeeper notification issue (#924)
```
The 904 is what we need and 785/924 help Ambry as well.

File: ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixAdmin.java (hunk 1 above)
Comment (jsjtzyy) [reply ambry-clustermap/src/test/java/com/github/ambry/clustermap/MockHelixAdmin.java:605]: My bad, will add it to these methods.

**Pull request merged**